### PR TITLE
Marian Progress Bar

### PR DESCRIPTION
**Pull request by mmlumba**
This doesn't have the component description; it mainly has the React component.
References #1009

### CODE REVIEW

File: client/app/containers/StyleGuide/StyleGuideProgressBar.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      </p>
+      <ProgressBar
+        sections = {
+        [
+          {
+            activated: true,
+            title: '1. Shopping Cart'
+          },
+          {
+            activated: true,
```
Comment (shanear) [inline client/app/containers/StyleGuide/StyleGuideProgressBar.jsx]: It seems like it'd be simpler to implement the logic of which steps are "active" inside the component.

The logic is that each step before the active step looks active. Each step after the active step looks inactive.

That way we only need to define which step is active (or `current`), and leave the rest blank.

Something like this:
```
sections = {
[
  { title: '1. Shopping Cart' },
  { title: '2. Checkout' },
  { title: '3. Confirmation', current: true }
]
```

File: client/app/containers/StyleGuide/StyleGuideProgressBar.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+// components
+import ProgressBar from '../../components/ProgressBar';
+
+export default class StyleGuideProgressBar extends React.Component {
+
+/* eslint class-methods-use-this: ["error", { "exceptMethods": ["render"] }] */
```
Comment (shanear) [inline client/app/containers/StyleGuide/StyleGuideProgressBar.jsx]: instead of doing the eslint escape, just make this a function component. See `StyleGuideTables.jsx`

Comment (mmlumba) [inline client/app/containers/StyleGuide/StyleGuideProgressBar.jsx]: Would this go under `RenderFunctions.jsx`?

File: client/app/containers/EstablishClaimPage/EstablishClaimProgressBar.jsx [hunk 3]
```diff
@@ -8,15 +8,15 @@ export default class EstablishClaimProgressBar extends React.Component {
...
           title: '1. Review Decision'
         },
         {
-          activated: this.props.isRouteClaim,
+          current: this.props.isRouteClaim,
           title: '2. Route Claim'
         },
         {
-          activated: this.props.isConfirmation,
+          current: this.props.isConfirmation,
```
Comment (sharonwarner) [inline client/app/containers/EstablishClaimPage/EstablishClaimProgressBar.jsx]: This works, but I don't think it's clear enough. All three sections will have 'current' set to true on the last page. We should set a restriction that only one section can have current set to true.

File: client/app/components/ProgressBar.jsx [hunk 4]
```diff
@@ -7,9 +7,23 @@ export default class ProgressBar extends React.Component {
...
     } = this.props;
 
+    let currentSectionIndex;
+
+    this.props.sections.forEach((section, i) => {
+      if (section.current === true) {
+        currentSectionIndex = i;
+      }
+    });
+
```
Comment (sharonwarner) [inline client/app/components/ProgressBar.jsx]: 👍

File: app/models/task.rb [hunk 5]
```diff
@@ -141,7 +141,7 @@ def prepare_with_decision!
   def cancel!(feedback = nil)
     transaction do
       update!(comment: feedback)
-      review! if may_review?
+      # review! if may_review?
```
Comment (shanear) [inline app/models/task.rb]: ?

File: client/app/components/ProgressBar.jsx [hunk 6]
```diff
@@ -7,9 +7,23 @@ export default class ProgressBar extends React.Component {
       sections
     } = this.props;
 
+    let currentSectionIndex;
+
+    this.props.sections.forEach((section, i) => {
+      if (section.current === true) {
+        currentSectionIndex = i;
+      }
+    });
```
Comment (shanear) [inline client/app/components/ProgressBar.jsx]: findIndex might be cleaner here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

```
let currentSectionIndex = this.props.sections.findIndex((section) => section.current === true)
```

File: client/app/components/ProgressBar.jsx [hunk 7]
```diff
@@ -7,9 +7,23 @@ export default class ProgressBar extends React.Component {
...
+
     return <div className="cf-app-segment">
       <div className="cf-progress-bar">
         {sections.map((section, i) => {
+          if (i <= currentSectionIndex && i !== null) {
```
Comment (shanear) [inline client/app/components/ProgressBar.jsx]: why would `i` be `null`?

File: client/app/containers/StyleGuide/StyleGuideProgressBar.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      <StyleGuideComponentTitle
+        title="Progress Bar"
+        id="progress_bar"
+        link="StyleGuideProgressBar.jsx"
+      />
+      <p>
+        This text here is the placeholder description for the Progress Bar.
+      </p>
+      <ProgressBar
+        sections = {
```
Comment (shanear) [inline client/app/containers/StyleGuide/StyleGuideProgressBar.jsx]: maybe break this out into a var like we do with the other `StyleGuideX` components?

File: app/models/task.rb (hunk 5 above)
Comment (mmlumba) [inline app/models/task.rb]: Since we don't do reviews anymore for dispatch cancellations I removed/commented it out.

File: app/models/task.rb [hunk 9]
```diff
@@ -141,7 +141,6 @@ def prepare_with_decision!
   def cancel!(feedback = nil)
     transaction do
       update!(comment: feedback)
-      review! if may_review?
```
Comment (shanear) [inline app/models/task.rb]: why was this removed?

Comment (mmlumba) [inline app/models/task.rb]: We don't use a review state for canceled tasks anymore (and this line would break for rake tests on canceled tasks).
